src/libsync/bulkpropagatorjob.cpp

```diff
@@ -231,6 +231,26 @@ void BulkPropagatorJob::triggerUpload()
     }
 }
 
+void BulkPropagatorJob::checkPropagationIsDone()
+{
+    if (_items.empty()) {
+        if (!_jobs.empty()) {
+            // just wait for the other job to finish.
+            return;
+        }
+        if (!_pendingChecksumFiles.empty()) {
```

Comment:
@mgallien You could combine it with previous `if`:

`if (!_jobs.empty())` -> `if (!_jobs.empty() && !_pendingChecksumFiles.empty())`

Reply:
sure